src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/constructors.jl

```diff
@@ -265,13 +265,19 @@ end
 
 
 ####################################################
-# 6: Equality
+# 6: Equality and hash
 ####################################################
 
 function Base.:(==)(ac1::RationalEquivalenceClass, ac2::RationalEquivalenceClass)
     return toric_variety(ac1) === toric_variety(ac2) && iszero(polynomial(ac1-ac2))
 end
 
+function Base.hash(ac::RationalEquivalenceClass, h::UInt) 
+  b = 0xb5d4ac6b9084eb6e  % UInt
+  h = hash(toric_variety(ac), h)
+  return xor(h, b)
```

Comment:
Thanks for your remark. Unfortunately, your proposal does not work in the intended way.

Julia requires for its internal functionality (e.g. `Set`) to work properly, that the equality of hashes is not finer than the usual equality `==`.

For your 1. Idea, this is clearly violated since hashing via the julia default results in the most finest equivalence relation, but there is a coarses one defined via `==`.

For your 2. idea, I am not familiar enough with the implementation to make a reasonable statement here. As I understood it, there can be equal RationalEquivalenceClass objects with a different `p` (at least this is what I thought based on the implementation of `==`). If this is true, I cannot see a way of sensibly defining a hash function without first choosing a canonical representative of the class (if that is even possible; not my field of knowledge). If my assumption is wrong, I would suggest to change the implementation of `==` to check for equality of the `toric_variety` and of `p` and change `hash` according to your proposal.

Reply:
Thank you @lgoettgens !

Reply:
Thank you @lgoettgens . This looks good to me!

Reply:
I am working on a bugfix for the toric varieties (https://github.com/oscar-system/Oscar.jl/pull/2396). There is a good chance that there will be a merge conflict for toric line bundles. The new and "good" methods use `picard_class` to compare two line bundles:
```julia
function Base.:(==)(l1::ToricLineBundle, l2::ToricLineBundle)
    return toric_variety(l1) === toric_variety(l2) && picard_class(l1) == picard_class(l2)
end
```
In other words, `divisor_class(...)` was/is to be replaced by `picard_class(...)`.